parser/prism/Translator.cc

```diff
@@ -126,12 +126,8 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             return make_unique<parser::Pair>(parser.translateLocation(loc), std::move(key), std::move(value));
         }
         case PM_ASSOC_SPLAT_NODE: {
-            auto assocSplatNode = reinterpret_cast<pm_assoc_splat_node *>(node);
-            pm_location_t *loc = &assocSplatNode->base.location;
-
-            auto value = translate(assocSplatNode->value);
-
-            return make_unique<parser::Kwsplat>(parser.translateLocation(loc), std::move(value));
+            unreachable("PM_ASSOC_SPLAT_NODE is handled separately in `Translator::translateHash()`, because its "
+                        "translation depends on whether its used in a hash or in a method call");
```

Comment:
This prevents us from needing to maintain a parsing context, that remembers whether this recursive call to `Translator::translate()` originated from a `PM_HASH_NODE` or a `PM_KEYWORD_HASH_NODE`.